### PR TITLE
Provide a map of action to state in WASM

### PR DESCRIPTION
**Pull request by andybalaam**

### CODE REVIEW

File: platforms/web/lib/useListeners/event.ts [hunk 1]
```diff
@@ -146,7 +96,12 @@ export function handleInput(
     testUtilities: TestUtilities,
     formattingFunctions: FormattingFunctions,
     inputEventProcessor?: InputEventProcessor,
-) {
+):
+    | {
+          content: string | undefined;
```
Comment (florianduros) [inline platforms/web/lib/useListeners/event.ts:101]: `content?: string`

**Comment by florianduros** [comment]
In the JS part, nothing to say :)

File: platforms/web/lib/useListeners/event.ts (hunk 1 above)
Comment (andybalaam) [reply platforms/web/lib/useListeners/event.ts]: Done, and also I added a rename, which I'd be grateful if you could review too please @florianduros

File: platforms/web/lib/useListeners/utils.ts [hunk 2]
```diff
@@ -14,14 +14,29 @@ See the License for the specific language governing permissions and
...
-import { FORMATTING_ACTIONS } from '../constants';
-import { FormattingState, FormattingActions } from '../types';
+import { ACTION_TYPES } from '../constants';
+import { ActionTypes, ActionState, AllActionStates } from '../types';
 
-export function getDefaultFormattingStates() {
-    return FORMATTING_ACTIONS.reduce<
-        Record<FormattingActions, FormattingState>
-    >((acc, action) => {
+export function defaultActionStates(): AllActionStates {
```
Comment (florianduros) [inline platforms/web/lib/useListeners/utils.ts:20]: I prefer when the functions have a verb in their name to be more explicit

**Review by florianduros — CHANGES_REQUESTED** [review]

File: platforms/web/lib/useListeners/utils.ts (hunk 2 above)
Comment (florianduros) [reply platforms/web/lib/useListeners/utils.ts:20]: Yes but syntactically it's a function and you can't use it as `const myState = defaultActionStates`

Comment (andybalaam) [reply platforms/web/lib/useListeners/utils.ts]: Cool, renamed to createDefaultActionStates

**Review by florianduros — APPROVED** [review]

**Pull request merged**